diff --git a/src/core/main.c b/src/core/main.c
index 94602611a..8779eef56 100644
--- a/src/core/main.c
+++ b/src/core/main.c
@@ -1531,9 +1531,11 @@ int main(int argc, char *argv[]) {
                  * need to do that for user instances since they never log
                  * into the console. */
                 log_show_color(colors_enabled());
-                r = make_null_stdio();
-                if (r < 0)
-                        log_warning_errno(r, "Failed to redirect standard streams to /dev/null: %m");
+		if (detect_container() <= 0) {
+                	r = make_null_stdio();
+	                if (r < 0)
+        	                log_warning_errno(r, "Failed to redirect standard streams to /dev/null: %m");
+		}
         }
 
         r = initialize_join_controllers();
